src/JZSolver.cpp

```diff
@@ -0,0 +1,106 @@
+#include "JZSolver.hpp"
+#include <numeric>
+#include <iostream>
+
+bool JZSolver::solve(std::shared_ptr<Board> b)
```

Comment:
Please do not pass share_ptr by value, use &.

Reply:
done